src/sql/parts/notebook/notebookInput.ts

```diff
@@ -166,20 +205,38 @@ export class NotebookInput extends EditorInput {
 		return NotebookInput.ID;
 	}
 
-	public resolve(refresh?: boolean): TPromise<IEditorModel> {
-		return undefined;
+	getResource(): URI {
+		return this.resource;
 	}
 
-	public getName(): string {
-		if (!this._title) {
-			this._title = resources.basenameOrAuthority(this._model.notebookUri);
+	async resolve(): TPromise<NotebookEditorModel> {
+		if (this._model && this._model.isModelCreated()) {
+			return TPromise.as(this._model);
+		} else {
+			let textOrUntitledEditorModel: UntitledEditorModel | IEditorModel;
+			if (this.resource.scheme === Schemas.untitled) {
+				textOrUntitledEditorModel = await this._untitledEditorService.loadOrCreate({ resource: this.resource, modeId: notebookModeId });
+			}
+			else {
+				const textEditorModelReference = await this.textModelService.createModelReference(this.resource);
+				textOrUntitledEditorModel = await textEditorModelReference.object.load();
+			}
+			this._model = this.instantiationService.createInstance(NotebookEditorModel, this.resource, textOrUntitledEditorModel);
+			this._model.onDidChangeDirty(() => this._onDidChangeDirty.fire());
+			return this._model;
 		}
-
-		return this._title;
 	}
 
-	public get isTrusted(): boolean {
-		return this._model.isTrusted;
+	private assignProviders(): void {
```

Comment:
Oh does this automatically handle provider assignment? No need to pass in from elsewhere? Great! Then only connectionProfileId is untracked / missed in this iteration?